pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -293,30 +295,67 @@ func (c *Cluster) Start() error {
 		return errors.Wrapf(err, "failed to update ceph storage status")
 	}
 
-	if c.spec.Storage.Store.UpdateStore == OSDStoreUpdateConfirmation {
-		delOpts := &k8sutil.DeleteOptions{WaitOptions: k8sutil.WaitOptions{Wait: true}}
-		err := k8sutil.DeleteConfigMap(c.clusterInfo.Context, c.context.Clientset, OSDReplaceConfigName, namespace, delOpts)
+	logger.Infof("finished running OSDs in namespace %q", namespace)
+	return nil
+}
+
+func (c *Cluster) startOSDMigration() (*migrationConfig, error) {
+	if !c.isMigrationRequested() {
+		logger.Debug("no OSD migration is requested")
+		return nil, nil
+	}
+
+	logger.Info("osd migration is requested")
+
+	// start migration only if PGs are active+clean
+	pgsHealhty, err := c.waitForHealthyPGs()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to wait for pgs to be healthy")
+	}
+
+	if !pgsHealhty {
+		return nil, errors.Wrapf(err, "failed to start migration due to unhealthy PGs")
+	}
+
+	// skip migration if previously migrated OSD is not up yet.
+	migrationComplete, err := isLastOSDMigrationComplete(c)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to check if the last migration was successful or not")
+	}
+
+	if !migrationComplete {
+		return nil, errors.Wrapf(err, "migration of the last OSD is not complete")
+	}
+
+	mc, err := c.newMigrationConfig()
```

Comment:
`mc` isn't a common acronym
```suggestion
	migrationConfig, err := c.newMigrationConfig()
```

Reply:
Done.